migrations/20231226012200_shared_modpacks.sql

```diff
@@ -0,0 +1,45 @@
+CREATE TABLE shared_profiles (
+    id bigint PRIMARY KEY,
+    name varchar(255) NOT NULL,
+    owner_id bigint NOT NULL,
+    icon_url varchar(255),
+    color integer NULL,
+    updated timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    created timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    loader_id int NOT NULL REFERENCES loaders(id), 
+    metadata jsonb NOT NULL DEFAULT '{}'::jsonb,
+
+    game_id int NOT NULL REFERENCES games(id),
+    game_version_id int NULL REFERENCES loader_field_enum_values(id) -- Minecraft java
+
+);  
+
+CREATE TABLE shared_profiles_mods (
+    shared_profile_id bigint NOT NULL REFERENCES shared_profiles(id),
+
+    -- for versions we have hosted
+    version_id bigint NULL REFERENCES versions(id), -- for versions
+
+    -- for cdn links to files we host directly
+    file_hash varchar(255) NULL,
```

Comment:
We should restructure this. Override files should be a separate table which this table refers too IMO. And then it should refer to this here (we could just re-use the files table TBH).

Like other API objects, this would allow us to have SHA1 and SHA512 hashes. Shared modpacks will also share files a ton (ex: if someone has the same exact config file between packs, we shouldn't haave to host it twice or have duplicate database entries)

Reply:
I think just using the files table would probably work well enough. It would also allow it so if someone did ever upload their mod to Modrinth, we could "re-link" it to the project dynamically and change it from an override to a Modrinth version ID. If that makes sense